### PR TITLE
update table to reflect secret required

### LINKED ISSUE #2259: update to client secrets table
Page:
- https://site-local.fusionauth.io/docs/v1/tech/oauth/endpoints#client-secret-table

The table used to describe when the `client_secret` is required isn't quite correct. 

1. Instead of calling out where the `client_secret` is required - body or header, we should just say it is required. The `client_secret` can be provided in the Basic auth header, or in the request body. 
2. You can always send PKCE, so instead of saying `N/A` shouldn't we say `yes` or `no` indicating if it is used by the caller?

### PR DESCRIPTION
**Pull request by alex-fusionauth**
Fixes #2259

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205076385141046

### CODE REVIEW

**Review by robotdan — COMMENTED** [review]
In this section we say:

>  If required, it may be placed in either the Authorization header or the request body.

In practice it is a bit more complicated than this, I'd rather just ask the user to follow the guidelines already documented in the endpoint doc. In each endpoint doc it describes when and how to use the `client_secret` in the request body or the `Authorization` header using HTTP Basic Auth. 

> Even when the value is not required, if it is provided in the request, it must be valid.


Can we say this instead? 

> Even when the client secret is not required, if it is provided it will be validated. 


Can we rename the column `clientAuthenticationPolicy value` to `clientAuthenticationPolicy`, I don't think adding the `value` suffix is necessary.

The `Client Credentials Grant` does not belong in this table. It is does not use `clientAuthenticationPolicy`.

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 1]
```diff
@@ -2051,110 +2051,66 @@ This parameter is only returned during an HTTP redirect if it was provided on th
...
-|N/A
-|N/A
-|N/A
-
-|Password Credentials Grant
-|`Required`
-|N/A
-|Yes
-|No
+|-
```
Comment (lyleschemmerling) [inline site/docs/v1/tech/oauth/endpoints.adoc:2091]: I feel like the blanks are confusing here. I don't really know what they mean. I feel that the N/A was more descriptive

**Review by lyleschemmerling — APPROVED** [review]
I don't feel super strongly one way or the other about the N/As but I do feel like omitting them is less direct about the intent of a column value.

Do mind that these docs are under migration so let us know when this merges so that we can rework it.

File: site/docs/v1/tech/oauth/endpoints.adoc (hunk 1 above)
Comment (robotdan) [reply site/docs/v1/tech/oauth/endpoints.adoc:2091]: I want to use `-`. This is a consistent pattern we have in docs and in the UI. If we want to add a footer that indicates `-`  implies not applicable we can do that.

**Comment by mooreds** [comment]
@alex-fusionauth can you please take a todo to update DocsDevReadme to capture this: https://github.com/FusionAuth/fusionauth-site/pull/2560#discussion_r1353014716

**Comment by alex-fusionauth** [comment]
Change is stale

**Pull request closed without merging**